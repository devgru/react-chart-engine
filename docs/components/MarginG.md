# MarginG

Компонент рендерит `g`-элемент с отступом слева и сверху.
Используется для сдвига системы координат вложенных элементов.

## Свойства

- число `top`, отступ сверху;
- число `left`, отступ слева.
