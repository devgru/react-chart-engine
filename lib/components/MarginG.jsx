import React from 'react';
import PropTypes from 'prop-types';

function MarginG({ left, top, children }) {
  return (
    <g transform={`translate(${left},${top})`}>
      {children}
    </g>
  );
}

MarginG.propTypes = {
  children: PropTypes.node,

  left: PropTypes.number,
  top: PropTypes.number,
};

MarginG.defaultProps = {
  children: null,

  left: 0,
  top: 0,
};

export default MarginG;
