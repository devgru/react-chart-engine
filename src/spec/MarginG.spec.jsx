import React from 'react';
import renderer from 'react-test-renderer';

import MarginG from '../lib/components/MarginG';

describe('MarginG', () => {
  it('should initialize at all', () => {
    const component = renderer.create(
      <MarginG>
        <circle />
      </MarginG>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should work with props', () => {
    const component = renderer.create(
      <MarginG top={10} left={10}>
        <circle />
      </MarginG>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
