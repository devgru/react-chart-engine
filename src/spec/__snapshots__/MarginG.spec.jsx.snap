// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarginG should initialize at all 1`] = `
<g
  transform="translate(0,0)"
>
  <circle />
</g>
`;

exports[`MarginG should work with props 1`] = `
<g
  transform="translate(10,10)"
>
  <circle />
</g>
`;
